src/compiler/types.ts

```diff
@@ -2748,6 +2756,8 @@ namespace ts {
     export interface AnonymousType extends ObjectType {
         target?: AnonymousType;  // Instantiation target
         mapper?: TypeMapper;     // Instantiation mapper
+        elementType?: Type;      // Element expressions of evolving array type

```

Comment:
Oh, I see. The issue with doing it that way is that we don't have a `TypeFlags.EvolvingArrayType` that would indicate an evolving array type (because we're out of flag bits). Instead, we distinguish by looking for the `elementType` property on `AnonymousType`, so it has to be part of `AnonymousType`.